### PR TITLE
[python] masking support for numeric input

### PR DESCRIPTION
**Pull request by ronie**
while the option to mask text input is available in python (ie. when entering a password), it was lacking support for numeric input.

this PR adds numeric masking support to:
- edit controls
```
self.edit = xbmcgui.ControlEdit(100, 250, 125, 75, 'Password')
self.edit.setType(xbmcgui.INPUT_TYPE_PASSWORD_NUMBER_VERIFY_NEW)
```

- dialog numeric
```
xbmcgui.Dialog().numeric(4, 'Enter password')
```

nummask

@MilhouseVH

### CODE REVIEW

**Comment by MilhouseVH** [comment]
Thanks @ronie - we'll give it a test!

File: xbmc/interfaces/legacy/Control.h [hunk 1]
```diff
@@ -1068,22 +1068,24 @@ namespace XBMCAddon
...
+      /// |-----------------------------------------------|:--------------------------------------------|
+      /// | xbmcgui.INPUT_TYPE_TEXT                       | (standard keyboard)
+      /// | xbmcgui.INPUT_TYPE_NUMBER                     | (format: #)
+      /// | xbmcgui.INPUT_TYPE_DATE                       | (format: DD/MM/YYYY)
+      /// | xbmcgui.INPUT_TYPE_TIME                       | (format: HH:MM)
+      /// | xbmcgui.INPUT_TYPE_IPADDRESS                  | (format: #.#.#.#)
+      /// | xbmcgui.INPUT_TYPE_PASSWORD                   | (input is masked)
+      /// | xbmcgui.INPUT_TYPE_PASSWORD_MD5               | (input is masked, return md5 hash of input)
+      /// | xbmcgui.INPUT_TYPE_SECONDS                    | (format: SS or MM:SS or HH:MM:SS or MM min)
+      /// | xbmcgui.INPUT_TYPE_PASSWORD_NUMBER_VERIFY_NEW | (numeric input is masked)
```
Comment (enen92) [inline xbmc/interfaces/legacy/Control.h:1081]: Can you explain why the "verify new" in the constant name? Why not simply INPUT_TYPE_NUMBER_MASKED?

**Comment by MilhouseVH** [comment]
Many thanks for this, @ronie, however this PR isn't quite what I had in mind.

The `type=4` numeric dialog is implementing the new PIN "set" logic (ie. enter a "new" password and then "verify" the same password by entering it a second time), when (as @enen92 suggests) we actually only want to mask the PIN of the user as they attempt to unlock the PIN restricted resource. 

It's entirely possible `type=4` as implemented by this PR would be useful in some scenario, but would it be possible to mask the single-entry numeric dialog?

Maybe a `type=5` could be used to accept masked numeric input (without verify). If it can allow the heading to be configured as well that would be gravy! :)

Currently we have this dialog (unmasked):
s1

File: xbmc/interfaces/legacy/Control.h (hunk 1 above)
Comment (ronie) [reply xbmc/interfaces/legacy/Control.h:1081]: these are 1:1 mappings to:
https://github.com/xbmc/xbmc/blob/d3a641fe5132cb6b2ea016e2a2c1d8eaf794795e/xbmc/guilib/GUIEditControl.h#L26-L42

guess the 'verify' bit relates to the fact you need to enter a password twice when setting it,
the second time is done to verify you didn't make a typo.

Comment (MilhouseVH) [reply xbmc/interfaces/legacy/Control.h:1081]: I guess what I really need, which I suspect is going to be more work, is a new `INPUT_TYPE_PASSWORD_NUMBER` that would accept only masked digits and that would return the entered digits. Ideally with a configurable heading. :smile:

**Comment by CastagnaIT** [comment]
i agree with MilhouseVH,
similar case for the PIN, is in VOD addons,
you do not know the value to verify, the "verify" is demanded to VOD service server that verify the PIN entered and return if it is valid or not

so a masked numeric input (without verify) is required, would be the most useful method

File: xbmc/interfaces/legacy/Control.h (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/interfaces/legacy/Control.h:1081]: Maybe remove then direct on description text the `(...)`, think is from time where it was not included inside a table:
```
xbmcgui.INPUT_TYPE_TEXT (standard keyboard)
```

**Comment by ronie** [comment]
thanx @MilhouseVH / @CastagnaIT i now get what you were looking for.

i've added another option to mask regular numeric input:
`pwd = xbmcgui.Dialog().numeric(0, 'Enter PIN', bHiddenInput=True)`

please test and let me know if i got it right this time ;-)

**Comment by MilhouseVH** [comment]
Thanks @ronie - this PR is working perfectly now for my use-case!

**Comment by MilhouseVH** [comment]
I'll include this for testing in my LibreELEC nightly builds, assuming it doesn't merge beforehand. Thanks again!

**Comment by lrusak** [comment]
jenkins build this please

**Comment by CastagnaIT** [comment]
Tested is working good thanks!

**Comment by lrusak** [comment]
jenkins build this please

**Comment by ronie** [comment]
jenkins build this please

**Comment by MilhouseVH** [comment]
Thanks @ronie!

**Pull request merged**